operators/src/adapters/feature_collection_merger.rs

```diff
@@ -295,4 +295,54 @@ mod tests {
             .unwrap()
         );
     }
+
+    #[tokio::test]
+    async fn interleaving_pendings() {
+        let mut stream_history: Vec<Poll<Option<Result<MultiPointCollection>>>> = vec![
+            Poll::Pending,
+            Poll::Ready(Some(
+                MultiPointCollection::from_data(
+                    MultiPoint::many(vec![(0.0, 0.1)]).unwrap(),
+                    vec![TimeInterval::new(0, 1).unwrap()],
+                    Default::default(),
+                )
+                .map_err(Error::from),
+            )),
+            Poll::Pending,
+            Poll::Ready(Some(
+                MultiPointCollection::from_data(
+                    MultiPoint::many(vec![(1.0, 1.1)]).unwrap(),
+                    vec![TimeInterval::new(0, 1).unwrap()],
+                    Default::default(),
+                )
+                .map_err(Error::from),
+            )),
+        ];
+        stream_history.reverse();
+
+        let stream = futures::stream::poll_fn(move |cx| {
+            let item = stream_history.pop().unwrap_or(Poll::Ready(None));
+
+            if let Poll::Pending = item {
+                cx.waker().wake_by_ref();
```

Comment:
I think this might cause problems in multithreaded executors, as the caller is woken up before the Pending-poll arrives. Either make sure that it works in multithreaded executors or ensure that this test is run with single threaded executor (don't run as tokio::test and create a new scheduler

Reply:
dec39c6